src/classes/VOL_CTRL_PersonalSiteContactInfo.cls

```diff
@@ -279,26 +279,30 @@ global with sharing class VOL_CTRL_PersonalSiteContactInfo {
         return null;
     }
 
-    global PageReference save() { 
-        if (contactReadOnly != null) {          
-            Set<String> contactFieldsToUpdate = new Set<String>();
+    global PageReference save() {
+        try {
+            if (contactReadOnly != null) {
+                Set<String> contactFieldsToUpdate = new Set<String>();
+
+                //  now copy over all the non-null fields from the form's contact to the existing contact.
+                for (string strF : listStrContactFields) {
+                    if (contactEdit.get(strF) != null) {
+                        try {
+                            contactReadOnly.put(strF, contactEdit.get(strF));
+                            contactFieldsToUpdate.add(strF);
+                        } catch (exception ex) {
 
-            //  now copy over all the non-null fields from the form's contact to the existing contact.
-            for (string strF : listStrContactFields) {
-                if (contactEdit.get(strF) != null) {
-                    try {
-                        contactReadOnly.put(strF, contactEdit.get(strF));
-                        contactFieldsToUpdate.add(strF);
-                    } catch(exception ex) {
-                        
+                        }
                     }
                 }
-            }
 
-            VOL_SharedCode.checkUpdateAccessSites('Contact', contactFieldsToUpdate);
-            update contactReadOnly;
+                VOL_SharedCode.checkUpdateAccessSites('Contact', contactFieldsToUpdate);
+                update contactReadOnly;
+            }
+            isEditing = false;
+        } catch (Exception ex) {
```

Comment:
May want to use a different exception variable (ex) for this catch().

Reply:
why?  it is in different scope and is the commonly used name for this variable.

Reply:
Both try/catch statements in the save method catch the generic Exception. It would be great to handle specific exceptions when possible, but in lieu of that having different variable names would allude to the differing exception handling cases -- 1. Read-only formula fields and 2. Security check/DML statement.

Reply:
I'm afraid I disagree with you.  If a common variable is used within its own scope, I always use the common name, and don't try to create unique names so they aren't repeated within the method.  for example, if a method had multiple for loops that iterated over contacts, i would most always use "for (Contact con : contacts)".  So I believe this is a personal preference, and I'm not comfortable making this change.  thanks.

Reply:
A usage of a common variable name as you mentioned in your example is fine. However, the usage of catching the generic Exception in both try/catch statements is the focal point of my comment. Because both statements are handling the same exception in different ways it gives pause while reading the code.

I'm fine with leaving the code as is. The variable name change was a suggestion for adding readability.